index.js

```diff
@@ -1,5 +1,110 @@
 const battleship = () => {
-  return 'The winner is...?'
+  //step 1
+
+  //objects of players(works)
+  let player1 = {
+    name: prompt("Player 1, enter your name").toUpperCase(),
+    shipCount: 4,
+    gameBoard:[
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ]]
+    }
+
+    let player2 = {
+    name: prompt("Player 2, enter your name").toUpperCase(),
+    shipCount: 4,
+    gameBoard:[
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ]]
+
+  }
+
+
+  //step 2(works)
+
+  //setup of ships using the random coordinate placments for each players board.
+
+    let ships=0;
+    let xPos;
+    let yPos;
+
+    while(ships < player1.shipCount){
+      xPos = Math.floor(Math.random() * 4);
+      yPos = Math.floor(Math.random() * 4);
+        if(parseInt(player1.gameBoard[xPos][yPos])!==1){
+        player1.gameBoard[xPos][yPos] = 1;
+        ships++;
+        }
+
+    }
+
+    while(ships < player2.shipCount){
+      xPos = Math.floor(Math.random() * 4);
+      yPos = Math.floor(Math.random() * 4);
+        if(parseInt(player2.gameBoard[xPos][yPos])!==1){
+        player2.gameBoard[xPos][yPos] = 1;
+        ships++;
+        }
+    }
+
+
+
+
+  //console.log(player1.gameBoard.toString())
+
+
+  //step 3
+  // play game. Used a remainder to switch between players. conditional checks if players coordinants match random
+
+  //function playGame(){
+    let currentPlayer;
+    let otherPlayer;
+    let count=0;
+
+      while(true){
+        if(+(count%2)===0){
+        currentPlayer = player1;
+        otherPlayer = player2;
+        }else{
+        currentPlayer = player2;
+        otherPlayer = player1;
+        }
+
+        let qxp = prompt(`${currentPlayer.name} Please choose your x coordinate`);
+        let qyp = prompt(`${currentPlayer.name} Please choose your y coordinate`);
+
+        if(currentPlayer.gameBoard[qxp][qyp]===1){
```

Comment:
You should be checking the opponent's board here instead of the current player's. Each turn a player is trying to guess where a ship is located on their opponent's board, not their own.

Reply:
yes! I was wondering about this. bc player1 was hitting their own board. ok this makes sense.